### PR TITLE
Dissident Scripts: Add thread management and compact embeds scripts

### PR DESCRIPTION
**Pull request by Shadowfiend**
Borrowed extensively from http://github.com/thesis/bishop, h/t @beaushinkle .

The thread management scripts are modified from their original form; in particular, rather than always mentioning the same role to ensure thread visibility, a heuristic is used:

- If the thread's channel has a role with the same name, that role is used.
- If not, but the thread's channel's category has a role with the same name, that role is used.
- If neither of these is true and the post is in #main in the General category, @-everyone is used.

Notably, Bishop's management of thread lifecycle, specifically around keeping threads alive for 7 days always, as well as pinging a thread to see if it should be archived periodically, are not yet included.

Also, the file organization is very different here compared to Bishop, keeping related functionality in a single file instead of spreading out over many files. This is intentional.

### CODE REVIEW

**Comment by Shadowfiend** [comment]
Lightly tested this myself, planning on turbo-merging once everything is green but would still love some review.

File: discord-scripts/compact-embeds.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { PartialMessage, Client, Message, EmbedBuilder } from "discord.js"
+
+async function compactGithubEmbeds(message: Message<boolean> | PartialMessage) {
+  if (message.author === null || message.author === undefined) {
+    return
+  }
+
+  if (!message.author.bot) {
```
Comment (beaushinkle) [inline discord-scripts/compact-embeds.ts:8]: we can remove a level of nesting by changing this to

```typescript
if (message.author.bot) {
  return
}
```

I became convinced of the value of this after watching https://www.youtube.com/watch?v=CFRhGnuXG-4

File: discord-scripts/thread-management.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
+import { Channel, ChannelType, Client } from "discord.js"
+
+// Emoji used to suggest a thread.
+const THREAD_EMOJI = "🧵"
+// Category that is treated as recreational, i.e. the rules don't apply baby.
+const RECREATIONAL_CATEGORY_ID = "1079492118692757605"
```
Comment (beaushinkle) [inline discord-scripts/thread-management.ts:6]: I find it really helpful to set this kind of stuff as environment variables, that way I can have my local testbot run on a test server

image

File: discord-scripts/thread-management.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    if (categoryMatchingRole !== undefined) {
+      await placeholder.edit(categoryMatchingRole.toString())
+      return
+    }
+
+    if (
+      categoryChannel?.name?.toLowerCase()?.endsWith("general") === true &&
+      containingChannel?.name?.toLowerCase()?.endsWith("main") === true
+    ) {
+      await placeholder.edit(server.roles.everyone.toString())
```
Comment (beaushinkle) [inline discord-scripts/thread-management.ts:83]: is this @-everyone? if so, i don't think the @-everyone trick works; have to give everyone the same role and use that instead

Comment (Shadowfiend) [reply discord-scripts/thread-management.ts:83]: Does "not work" mean people don't get joined correctly? Might explain the issue @michalinacienciala noticed @ https://discord.com/channels/1079490158639976609/1090503843508408330/1090642962729414656 .

Comment (beaushinkle) [reply discord-scripts/thread-management.ts:83]: yeah - as far as I know, editing into @-everyone doesn't add folks

Comment (beaushinkle) [reply discord-scripts/thread-management.ts:83]: @-everyone:
image

@-keep:
image

**Pull request merged**